config/packages/github_api.yaml

```diff
@@ -1,4 +1,10 @@
 services:
+    _defaults:
+        # The Github\Client is only calling authenticate/addCache when public (but I don't know why)
```

Comment:
Nice catch, can you open an issue on the repo and ask?

Reply:
Should it be on the Github\Client repo or on the symfony repo ?

Reply:
I'm curious why that would happen, just making `Github\Client` public fixes the issue or should all services defined there be public?

I've never used the commands.

Reply:
I just discovered that I don't need to add
```
    _defaults:
        # The Github\Client is only calling authenticate/addCache when public (but I don't know why)
        public: true
        autowire: true
        autoconfigure: true
```

If I remove the call 
```
- ['addCache', ['@Cache\Adapter\Redis\RedisCachePool']]
```

Maybe it's related to the failure I get
```
                                                                                                                        
 [WARNING] Some commands could not be registered:                                                                       
                                                                                                                        

In getClientService.php line 53:
                      
  Connection refused  
```

The failure of one calls is denying every calls ? how to solve this ?

Reply:
@sonata-project/contributors Any idea why the `addCache` call is disabling the github authentication ?

@OskarStark should we remove it ?

Reply:
https://github.com/KnpLabs/php-github-api/issues/947

Maybe this helps?

Reply:
Ok I found the reason.

When I correctly configure `redis` the `authenticate` call is made.
So it was related to the error
```
 [WARNING] Some commands could not be registered:                                                                       
                                                                                                                        

In getClientService.php line 53:
                      
  Connection refused  
```

Is there a way to add the `addCache` call only if redis server is started ? ^^'